index.bs

```diff
@@ -514,12 +507,23 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
 
 <div algorithm>
 To <dfn>respond with an error</dfn> given a [=WebSocket connection=]
-|connection| and an [=error code=] |code|:
+|connection|, |command id| and |error code|:
 
- 1. Issue: Form a valid JSON |errorObject| given |code|.
+ 1. If |command id| is not a number, let |command id| be null.
```

Comment:
Our handling of invalid id seems a bit broken/incomplete still. If the number wasn't in the CDDL `uint` range, should we send back that number, or null? I think null would make sense, but I don't think spec makes it so.

This algorithm ("respond with an error") is called by "handle an incoming message", but that just parses JSON and doesn't try to interpret the `id` member. Then we have "Match data against the remote end definition" in "process a command" which operates on the unparsed scalar value string, and that would fail if the id isn't a CDDL `uint`. However, it doesn't extract a known-to-be-uint id if something else in the schema didn't match.

Overall, that we both parse the message payload as JSON and interpret it per CDDL seems suspect, and probably not how one would implement this.

If fixing this in this PR is too much, then a follow-up issue would be OK.